Implement FoldConvBatchnorm2d pass.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#25282 Implement FoldConvBatchnorm2d pass.**

For now it will be used in quantization, but it can be used as a
standalone pass too.

Couple of things are not finished at this moment:
- Batchnorm.eps value is hardcoded. This is bad and wrong, but we cannot
access fields listed in __constants__ from IR now. Once we fix this, we
should remove the hardcoded value.
- We do not remove Batchnorm submodules from the parent module even when
they were merged into a Conv. Once we figure out API for removing
attributes and modules, we should fix this.

Differential Revision: [D17086611](https://our.internmc.facebook.com/intern/diff/D17086611)

Commit:
Implement FoldConvBatchnorm2d pass.

For now it will be used in quantization, but it can be used as a
standalone pass too.

Couple of things are not finished at this moment:
- Batchnorm.eps value is hardcoded. This is bad and wrong, but we cannot
access fields listed in __constants__ from IR now. Once we fix this, we
should remove the hardcoded value.
- We do not remove Batchnorm submodules from the parent module even when
they were merged into a Conv. Once we figure out API for removing
attributes and modules, we should fix this.

Commit:
Lint on "Implement FoldConvBatchnorm2d pass."


For now it will be used in quantization, but it can be used as a
standalone pass too.

Couple of things are not finished at this moment:
- Batchnorm.eps value is hardcoded. This is bad and wrong, but we cannot
access fields listed in __constants__ from IR now. Once we fix this, we
should remove the hardcoded value.
- We do not remove Batchnorm submodules from the parent module even when
they were merged into a Conv. Once we figure out API for removing
attributes and modules, we should fix this.

Differential Revision: [D17086611](https://our.internmc.facebook.com/intern/diff/D17086611)

Commit:
clang-tidy on "Implement FoldConvBatchnorm2d pass."


For now it will be used in quantization, but it can be used as a
standalone pass too.

Couple of things are not finished at this moment:
- Batchnorm.eps value is hardcoded. This is bad and wrong, but we cannot
access fields listed in __constants__ from IR now. Once we fix this, we
should remove the hardcoded value.
- We do not remove Batchnorm submodules from the parent module even when
they were merged into a Conv. Once we figure out API for removing
attributes and modules, we should fix this.

Differential Revision: [D17086611](https://our.internmc.facebook.com/intern/diff/D17086611)

Commit:
clang-tidy on "Implement FoldConvBatchnorm2d pass."


For now it will be used in quantization, but it can be used as a
standalone pass too.

Couple of things are not finished at this moment:
- Batchnorm.eps value is hardcoded. This is bad and wrong, but we cannot
access fields listed in __constants__ from IR now. Once we fix this, we
should remove the hardcoded value.
- We do not remove Batchnorm submodules from the parent module even when
they were merged into a Conv. Once we figure out API for removing
attributes and modules, we should fix this.

Differential Revision: [D17086611](https://our.internmc.facebook.com/intern/diff/D17086611)

Commit:
Update on "Implement FoldConvBatchnorm2d pass."


For now it will be used in quantization, but it can be used as a
standalone pass too.

Couple of things are not finished at this moment:
- Batchnorm.eps value is hardcoded. This is bad and wrong, but we cannot
access fields listed in __constants__ from IR now. Once we fix this, we
should remove the hardcoded value.
- We do not remove Batchnorm submodules from the parent module even when
they were merged into a Conv. Once we figure out API for removing
attributes and modules, we should fix this.

Differential Revision: [D17086611](https://our.internmc.facebook.com/intern/diff/D17086611)

jerryzh168 reviewed (commented):
submitted reviews for tests first

jerryzh168 reviewed (changes requested):
Looks pretty good overall, please fix things mentioned in comments

Commit:
Update on "Implement FoldConvBatchnorm2d pass."


For now it will be used in quantization, but it can be used as a
standalone pass too.

Couple of things are not finished at this moment:
- Batchnorm.eps value is hardcoded. This is bad and wrong, but we cannot
access fields listed in __constants__ from IR now. Once we fix this, we
should remove the hardcoded value.
- We do not remove Batchnorm submodules from the parent module even when
they were merged into a Conv. Once we figure out API for removing
attributes and modules, we should fix this.

Differential Revision: [D17086611](https://our.internmc.facebook.com/intern/diff/D17086611)

jerryzh168 reviewed (approved):
Looks good, Thanks!

Commit:
Update on "Implement FoldConvBatchnorm2d pass."


For now it will be used in quantization, but it can be used as a
standalone pass too.

Couple of things are not finished at this moment:
- Batchnorm.eps value is hardcoded. This is bad and wrong, but we cannot
access fields listed in __constants__ from IR now. Once we fix this, we
should remove the hardcoded value.
- We do not remove Batchnorm submodules from the parent module even when
they were merged into a Conv. Once we figure out API for removing
attributes and modules, we should fix this.

Differential Revision: [D17086611](https://our.internmc.facebook.com/intern/diff/D17086611)
